ambry-network/src/main/java/com.github.ambry.network/SSLBlockingChannel.java

```diff
@@ -51,75 +48,38 @@ public SSLBlockingChannel(String host, int port, MetricRegistry registry, int re
   }
 
   @Override
-  public void connect() throws IOException {
-    synchronized (lock) {
-      if (!connected) {
-        Socket socket = new Socket();
-        socket.setSoTimeout(readTimeoutMs);
-        socket.setKeepAlive(true);
-        socket.setTcpNoDelay(true);
-        if (readBufferSize > 0) {
-          socket.setReceiveBufferSize(readBufferSize);
-        }
-        if (writeBufferSize > 0) {
-          socket.setSendBufferSize(writeBufferSize);
-        }
-        socket.connect(new InetSocketAddress(host, port), connectTimeoutMs);
-        sslSocket = (SSLSocket) sslSocketFactory.createSocket(socket, host, port, true);
+  public Socket createSocket() throws IOException {
+    Socket tcpSocket = super.createSocket();
```

Comment:
Could you call `tcpSocket.close()` if there was an exception during SSL initialization?

Reply:
Done (with SO_LINGER set to 0, to issue TCP RST instead of FIN).